### PR TITLE
feat: support for custom headers function 

### PR DESCRIPTION
**Pull request by ecaroth**
I am hosting gotenberg on cloud run as a private service, and for that I needed to include custom auth headers to validate my service account. There was no easy way to do that, so I added the capability to include a custom async function in the client config to generate/use any headers you might need to pass to the underlying `fetch` function at call time.

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +1 @@
-#!/usr/bin/env sh
-. "$(dirname -- "$0")/_/husky.sh"
```
Comment (ecaroth) [inline .husky/commit-msg:2]: required to remove for husky v9

**Comment by cherfia** [comment]
@ecaroth Thank you for your suggestion! Have you tried using the `extraHttpHeaders` parameter? For example, instead of:

```javascript
Chromiumly.configure({
  endpoint: "https://my-gotenberg-service-1234.us-central1.run.app",
  addCustomHeadersFn: async () => ({
    Authorization: `Bearer ${await generateIdToken()}`,
  }),
});
```
I'm assuming that you can achieve the same like this:
```javascript
import { HtmlConverter } from "chromiumly";

const htmlConverter = new HtmlConverter();
const buffer = await htmlConverter.convert({
  html: "path/to/index.html",
  extraHttpHeaders: { Authorization: `Bearer ${await generateIdToken()` },
});
```
This makes use of the built-in support for [custom HTTP headers](https://gotenberg.dev/docs/routes#custom-http-headers-chromium)

Please let me know if this approach work for you?

**Comment by ecaroth** [comment]
@cherfia I missed that completely -- my main use case was the LibreOffice converter, which doesn't currently support that param, though it appears adding support for it there might make the most sense

**Comment by ecaroth** [comment]
@cherfia Dug into this more `extraHeaders` is multipart form data that is passed to gotenberg, which it then uses when making _it's own_ requests. This MR I created allows you to add custom headers ad-hoc when calling gotenberg itself , which is an entirely different use case (and still required here!). I looked at some other gotenberg wrappers as well, which do something similar for just this reason.

**Review by cherfia — COMMENTED** [review]
@ecaroth Thanks again for looking into it! I made a few adjustments to align with the current codebase for consistency, but the results should remain the same. Could you please give it a try and let me know if it works for you?

**Comment by ecaroth** [comment]
@cherfia FYI I validated functionality is working as expected. Cheers!

**Pull request merged**